### DISCUSSION TITLE
Merging 10X Multiome (take two)

### DISCUSSION THREAD

**Discussion by mspivakov** [stuart-lab/signac#812] [Q&A] [answered]
Hello! 

You have recently addressed this topic (https://github.com/timoast/signac/discussions/790), but I have a slightly different situation.

**For the same pair of Multiome experiments, I ended up having two combined objects: one for RNA and one for ATAC.**

The one for RNA was merged from a list of SCT-processed individual objects and then batch-corrected by Harmony.

The one for ATAC was generated as described in your helpful vignette(https://satijalab.org/signac/articles/merging.html), with the exception that for multiome objects I had to filter by `atac_fragments>N & excluded_reason==0` rather than `passed_filters`.

I made sure cell names are identical between the objects:

```
> atac[["ATAC"]]
ChromatinAssay data with 60940 features for 7435 cells
Variable features: 60115 
Genome: 
Annotation present: TRUE 
Motifs present: FALSE 
Fragment files: 2 

>rna
An object of class Seurat 
55946 features across 7211 samples within 2 assays 
Active assay: SCT (19345 features, 2000 variable features)
 1 other assay present: RNA
 2 dimensional reductions calculated: pca, pca_umap
```
I then tried running:

`rna[["ATAC"]] = atac`

However, after much hanging, I get this error:

```
Error in `[<-.data.frame`(`*tmp*`, , i, value = new("Seurat", assays = list( : 
  replacement has 60940 rows, data has 7211
Calls: [[<- -> [[<- -> [<- -> [<-.data.frame
Execution halted

```

I also tried running simply: `merge(rna, atac) `, but for some strange reason this has messed up my RNA data (https://twitter.com/mikhailspivakov/status/1443873870507921423/photo/1).

What am I doing wrong and how can I make it right? :)

Many thanks!

**Update**: 

I should of course have been running `rna[["ATAC"]] = atac[["ATAC"]]`, which has initially failed as it requires the same number of cells, but subsetting for the same cells has fixed this. I am still wondering though whether it's to be expected that `merge(rna, atac)` didn't work?

**Comment by timoast** [comment, marked as answer]
`merge()` is for cases where you want to combine different cells into one object. Here you have different measurements for the same cells, so you want to add these as separate assays in the same object.
